lib/commands/implementations/voteban.js

```diff
@@ -32,53 +24,29 @@ function voteBan(ipBan, defaultBanTime, weighted) {
     }
     const { isPermanent, userToPunish, parsedDuration, parsedReason } = parsedInput;
 
-    voteBanStarted = true;
     const listener = services.messageRelay.startListenerForChatMessages('voteban');
     if (listener === false) {
       return new CommandOutput(null, 'Something went wrong?? uhh. Restart me.');
     }
+    listener.on('pollstop', (message) => {
+      const poll = JSON.parse(message);
 
-    const muteString = isPermanent
-      ? 'PERMANENTLY'
-      : formatDuration(moment.duration(parsedDuration, 'seconds'));
-    const votedMap = {};
-    let yes = 0;
-    let no = 0;
-
-    listener.on('message', (data) => {
-      if (votedMap[data.user]) {
-        return;
-      }
-      const message = data.message.trim();
-      if (message.trim() === '1' || message.trim() === '2') {
-        let votes = 1;
-        if (weighted) {
-          votes = _.max(
-            Object.keys(weightedTranslateMap).map((k) => {
-              const idx = data.roles.indexOf(k);
-              if (idx === -1) return 1;
-              return weightedTranslateMap[k];
-            }),
-          );
-        }
-        votedMap[data.user] = true;
-        // eslint-disable-next-line no-unused-expressions
-        message.trim() === '1' ? (yes += votes) : (no += votes);
-      }
-    });
-
-    setTimeout(() => {
-      voteBanStarted = false;
       services.messageRelay.stopRelay('voteban');
       services.messageRelay.sendOutputMessage('Total votes:');
+
+      const yes = poll.totals[poll.options.findIndex((option) => option === 'yes')];
+      const no = poll.totals[poll.options.findIndex((option) => option === 'no')];
```

Comment:
Lodash actually offers a pretty cute function called `zipObject` that might work here perfectly.